### PR TITLE
[Task][66485] Init API

### PR DESCRIPTION
**Pull request by Lomahs1401**
## Related Tickets
- [#Task 66485: Init API](https://edu-redmine.sun-asterisk.vn/issues/66485)
## WHAT
- Init Api and create RecyclerView
## Evidence (Screenshot or Video)

https://github.com/awesome-academy/FoodRecipeApp/assets/77973385/e5b8c270-50ef-4a46-b09a-9875b6d888b4

## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://sal.vn/WtYPAy |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://sal.vn/piPoQn |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/res/layout/fragment_home.xml [hunk 1]
```diff
@@ -2,19 +2,36 @@
...
+    <androidx.recyclerview.widget.RecyclerView
+        android:id="@+id/rcv_recipes"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_margin="@dimen/dp_4"
+        android:layout_marginStart="@dimen/dp_10"
+        android:layout_marginBottom="@dimen/dp_10"
+        app:layoutManager="androidx.recyclerview.widget.LinearLayoutManager"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/textView"
```
Comment (ghost) [inline app/src/main/res/layout/fragment_home.xml]:
```suggestion
        app:layout_constraintBottom_toBottomOf="parent"
        app:layout_constraintTop_toBottomOf="@id/textView"
```

File: app/src/main/res/layout/fragment_home.xml [hunk 2]
```diff
@@ -2,19 +2,36 @@
...
+        app:layout_constraintVertical_bias="0.027"
+        tools:layout_editor_absoluteX="4dp" />
```
Comment (ghost) [inline app/src/main/res/layout/fragment_home.xml]: dùng file dimen

File: gradle.properties [hunk 3]
```diff
@@ -21,7 +21,7 @@ kotlin.code.style=official
...
 # thereby reducing the size of the R class for that library
 android.nonTransitiveRClass=true
-BASE_URL=gefrtjsdf
-API_KEY=fhwerjklhwejrj
-BASE_URL_RECIPES=asdqwe
-BASE_URL_FOOD_INGREDIENTS=zxcasd
+BASE_URL=https://api.spoonacular.com/
+API_KEY=3985065635194ef7b79cc30353272c7a
+BASE_URL_RECIPES=recipes/
+BASE_URL_FOOD_INGREDIENTS=food/ingredients/
```
Comment (ghost) [inline gradle.properties]: sao lại đây hết thế này

Comment (Lomahs1401) [reply gradle.properties]: Done

File: app/src/main/res/layout/fragment_home.xml (hunk 2 above)
Comment (Lomahs1401) [reply app/src/main/res/layout/fragment_home.xml]: Done

File: app/src/main/res/layout/fragment_home.xml (hunk 1 above)
Comment (Lomahs1401) [reply app/src/main/res/layout/fragment_home.xml]: Done

File: app/src/main/java/com/example/foodrecipeapp/data/repo/source/remote/fetchjson/GetJsonFromUrl.kt [hunk 4]
```diff
@@ -1,12 +1,86 @@
...
+        }
+
+        val bufferedReader = BufferedReader(InputStreamReader(url.openStream()))
+        val responseData = StringBuilder()
+
+        bufferedReader.forEachLine {
+            responseData.append(it.trim())
+        }
+
+        Log.d(TAG, "Query Success: $responseData")
```
Comment (ghost) [inline app/src/main/java/com/example/foodrecipeapp/data/repo/source/remote/fetchjson/GetJsonFromUrl.kt]: bỏ hết log phục vụ debug đi e

Comment (Lomahs1401) [reply app/src/main/java/com/example/foodrecipeapp/data/repo/source/remote/fetchjson/GetJsonFromUrl.kt]: Done

**Review by ghost — APPROVED** [review]

**Pull request merged**